cephadm-preflight.yml

```
@@ -48,6 +48,205 @@
     - name: redhat family of OS related tasks
       when: ansible_facts['os_family'] == 'RedHat'
       block:
+        - name: Initialize os_check variable
+          set_fact:
+            os_check: { "failed": false }
+
+        - name: OS Preflight Checks
+          block:
+            - name: Initialize check results
+              set_fact:
+                preflight_results: []
+
+            - name: Ensure OS is RHEL 9+
+              fail:
+                msg: "Ceph requires RHEL 9+. Detected: {{ ansible_facts['distribution'] }} {{ ansible_facts['distribution_version'] }}"
+              when: 
+                - ansible_facts['distribution'] not in ['RedHat'] or ansible_facts['distribution_major_version'] | int < 9
+              register: os_check
+              ignore_errors: true
+
+            - name: Store OS check result
+              set_fact:
+                preflight_results: "{{ preflight_results + [{'Check': 'OS Version', 'Result': 'PASS' if ansible_facts['distribution_major_version'] | int >= 9 else 'FAIL'}] }}"
+
+            - name: Ensure SELinux is set to Enforcing mode
+              ansible.posix.selinux:
+                policy: targeted
+                state: enforcing
+              register: selinux_status
+              changed_when: false
+              failed_when: selinux_status.state != 'enforcing'
+
+            - name: Store SELinux check result
+              set_fact:
+                preflight_results: "{{ preflight_results + [
+                  {
+                    'Check': 'SELinux',
+                    'Result': 'PASS' if selinux_status.state == 'enforcing' else 'FAIL',
+                    'Reason': 'SELinux was not in enforcing mode and could not be enforced automatically' if selinux_status.failed else 'N/A'
+                  }
+                ] }}"
+
+            - name: Ensure required packages are installed
+              package:
+                name:
+                  - rpcbind
+                  - podman
+                  - firewalld
+                state: present
```

```suggestion
            - name: Ensure required packages are installed
              package:
                name:
                  - rpcbind
                  - podman
                  - firewalld
                state: present
```

we have a variable `infra_pkgs` in `ceph_defaults` for this

---

Using infra_pkgs in the current implementation.